src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -389,65 +369,35 @@ private static decimal GetRawDecimalConstant(CustomAttributeData attr)
         {
             Debug.Assert(attr.Constructor.DeclaringType == typeof(DecimalConstantAttribute));
 
-            foreach (CustomAttributeNamedArgument namedArgument in attr.NamedArguments)
-            {
-                if (namedArgument.MemberInfo.Name.Equals("Value"))
-                {
-                    // This is not possible because Decimal cannot be represented directly in the metadata.
-                    Debug.Fail("Decimal cannot be represented directly in the metadata.");
-                    return (decimal)namedArgument.TypedValue.Value!;
-                }
-            }
-
             ParameterInfo[] parameters = attr.Constructor.GetParameters();
             Debug.Assert(parameters.Length == 5);
 
             System.Collections.Generic.IList<CustomAttributeTypedArgument> args = attr.ConstructorArguments;
             Debug.Assert(args.Count == 5);
 
-            if (parameters[2].ParameterType == typeof(uint))
-            {
-                // DecimalConstantAttribute(byte scale, byte sign, uint hi, uint mid, uint low)
-                int low = (int)(uint)args[4].Value!;
-                int mid = (int)(uint)args[3].Value!;
-                int hi = (int)(uint)args[2].Value!;
-                byte sign = (byte)args[1].Value!;
-                byte scale = (byte)args[0].Value!;
-
-                return new decimal(low, mid, hi, sign != 0, scale);
-            }
-            else
-            {
-                // DecimalConstantAttribute(byte scale, byte sign, int hi, int mid, int low)
-                int low = (int)args[4].Value!;
-                int mid = (int)args[3].Value!;
-                int hi = (int)args[2].Value!;
-                byte sign = (byte)args[1].Value!;
-                byte scale = (byte)args[0].Value!;
-
-                return new decimal(low, mid, hi, sign != 0, scale);
-            }
+            // constructor DecimalConstantAttribute(byte scale, byte sign, uint hi, uint mid, uint low) is not CLS-compliant
+            bool isClsCompliant = parameters[2].ParameterType == typeof(int);
+
+            int low = isClsCompliant ? (int)args[4].Value! : (int)(uint)args[4].Value!;
+            int mid = isClsCompliant ? (int)args[3].Value! : (int)(uint)args[3].Value!;
+            int hi = isClsCompliant ? (int)args[2].Value! : (int)(uint)args[2].Value!;
+            byte sign = (byte)args[1].Value!;
+            byte scale = (byte)args[0].Value!;
+
+            return new decimal(low, mid, hi, sign != 0, scale);
```

Comment:
```suggestion
            return new decimal(low: Get(args, 2), mid: Get(args, 3), high: Get(args, 4),
                isNegative: ((byte)args[1].Value!), scale: (byte)args[0].Value!);
            
            static int GetInt32(IList<CustomAttributeTypedArgument> args, int index)
            {
                // The constructor is overloaded to accept both signed and unsigned arguments
                object obj = args[index].Value!;
                return (obj is int value) ? value : (uint)obj;
            }
```

> not CLS-compliant

CLS compliance is not directly relevant to this logic.